handlers/features.go

```diff
@@ -93,3 +93,79 @@ func (oh *featureHandler) GetFeatureByUuid(w http.ResponseWriter, r *http.Reques
 	w.WriteHeader(http.StatusOK)
 	json.NewEncoder(w).Encode(workspaceFeature)
 }
+
+func (oh *featureHandler) CreateOrEditFeaturePhase(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	pubKeyFromAuth, _ := ctx.Value(auth.ContextKey).(string)
+	if pubKeyFromAuth == "" {
+		fmt.Println("no pubkey from auth")
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	newPhase := db.FeaturePhase{}
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&newPhase)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "Error decoding request body: %v", err)
+		return
+	}
+
+	if newPhase.Uuid == "" {
+		newPhase.Uuid = xid.New().String()
+	}
+
+	if newPhase.CreatedBy != "" {
```

Comment:
@MahtabBukhari  Make a call to the DB to get the phase, before performing the created and createdBy conditions, the createdBy won't be sent during an update, which will make that condition always true.